src/pyosrd/agents/scheduler_agent.py

```diff
@@ -184,10 +175,23 @@ def regulate_scenario(
 
     def regulate_scenarii(
         self,
-        scenarii: list[str]
+        scenarii: list[str],
+        indicator_function: Callable[[Schedule, Schedule, OSRD], pd.DataFrame],
     ) -> pd.DataFrame:
         """Regulates a list of scenarii using a given agent.
 
+        A callable is required instead of direct weight ponderation
+        because this function is used to compute delays for
+        different simulations in regulate method of SchedulerAgent.
+        Therefore we need to be able to compute different weights
+        data frame based on the simulations.
+
+        Parameters
+        ----------
+        ref_schedule: Schedule
+            The reference schedule used as the ideal schedule
+        delayed_schedule: Schedule
+            The delayed schedule, regulated use to compute the indicator
         Parameters
```

Comment:
Update doctring with a description of `indicator_function`

Example(s) would be welcome